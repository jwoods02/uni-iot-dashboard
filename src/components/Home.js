import React, { Component } from 'react';
import '../App.css';
import Image from '../Photos/dww.jpg';

class Home extends Component {
  render() {
    return(
      <div id="homeWrapper">
        <section className="homeRow">
          <img src={Image} alt="Data Works Wales logo"/>
        </section>
        <section className="homeRow">
          <button className="btn btn-success">Community Login</button>
          <button className="btn btn-danger">Teaching Portal</button>
        </section>
      </div>
    )
  }
}

export default Home;
