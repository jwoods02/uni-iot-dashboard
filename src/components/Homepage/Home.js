import React, { Component } from 'react';
import '../../App.css';
import Image from '../../Assets/dww.jpg';
import { Link } from 'react-router-dom';

class Home extends Component {
  render() {
    return(
      <div id="homeWrapper">
        <section className="homeRow">
          <img src={Image} alt="Data Works Wales logo"/>
        </section>
        <section className="homeRow">
          <Link to="/login"><button className="btn btn-success">Community Login</button></Link>
          <button className="btn btn-danger">Teaching Portal</button>
        </section>
      </div>
    )
  }
}

export default Home;
