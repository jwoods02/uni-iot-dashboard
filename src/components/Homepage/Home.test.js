import React from "react";
// enzyme allows me to render one component to standalone without context or the rest of the application
import {configure, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Home from "./Home.js";
import renderer from "react-test-renderer";

configure({adapter: new Adapter()});

describe("<Home />", () =>{

    let wrapper;
    // beforeEach takes a function as an arguement - it executes this function before every "it" jest test
    beforeEach(() => {
        wrapper = shallow(<Home/>);
    });

    it("should render the Home component", () =>
    {
        // The code below is the expectation of the test
        expect(wrapper.find(Home));
    });


    it("matches the snapshot", () => {
        const tree = renderer.create(<Home/>).toJSON();

        expect(tree).toMatchSnapshot()
    })

});
