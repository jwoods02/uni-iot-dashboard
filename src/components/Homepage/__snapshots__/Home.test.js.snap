// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches the snapshot 1`] = `
<div
  id="homeWrapper"
>
  <section
    className="homeRow"
  >
    <img
      alt="Data Works Wales logo"
      src="dww.jpg"
    />
  </section>
  <section
    className="homeRow"
  >
    <button
      className="btn btn-success"
    >
      Community Login
    </button>
    <button
      className="btn btn-danger"
    >
      Teaching Portal
    </button>
  </section>
</div>
`;
